src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -540,14 +540,25 @@ def _apply_cluster_overrides(
         overrides: dict[str, str],
         wheel_runner: str,
     ) -> dict:
+        # filter out overrides that point to non-existing key
```

Comment:
the `_apply_cluster_overrides` function is only meant for testing and is not documented at all. it took me a while to figure out what it is doing.

- in its current form, it simply overrides a job cluster with an interactive cluster id to reduce start up time
- for ext hms tests, we need to be able to override a job cluster with another job cluster, i.e. overriding the cluster key, not cluster id. this replacement is crude however, because if the override job cluster key is not defined, the workflow definition is invalid (not all workflows need all 3 clusters `main`, `tacl` and `table-migration`

take the below example
```
workflow: group_migration
job_clusters: main - 1 node
task 1: scan_group, cluster_key: main
```

if override is `{main:interactive_id}`, we will change `task 1` to `task 1: scan_group, cluster_id: interactive_id
if override is `{main:table-migration}`, we cannot change `task_1 to `task_1:scan_group, cluster_key: table-migration` as `table-migration` is not defined yet

Reply:
Can't we make a best guess when something is a cluster id or when it is a cluster key? Cluster id's follow the following regex `^\d{4}-\d{6}-\w{8}$` (I think). If that is a match, assume it is a cluster id, otherwise assume it is a cluster key

Reply:
I've switched to a different approach